cms/datasets/blocks.py

```diff
@@ -65,35 +62,32 @@ def clean(self, value: StreamValue, ignore_required_constraints: bool = False) -
 
 
 class DatasetStoryBlock(StreamBlock):
-    dataset_lookup = DatasetChooserBlock(
-        label="Lookup Dataset", template="templates/components/streamfield/dataset_link_block.html"
-    )
+    dataset_lookup = DatasetChooserBlock(label="Lookup Dataset")
     manual_link = ManualDatasetBlock(
         required=False,
         label="Manually Linked Dataset",
     )
 
-    class Meta:
-        template = "templates/components/streamfield/datasets_block.html"
-
     def clean(self, value: StreamValue, ignore_required_constraints: bool = False) -> StreamValue:
         cleaned_value = super().clean(value)
 
-        # Validate there are no duplicate datasets, including between manual and looked up datasets referencing the same
-        # URL path
+        # Validate there are no duplicate datasets,
+        # including between manual and looked up datasets referencing the same URL
 
-        # For each dataset URL path, record the indices of the blocks it appears in
-        url_paths = defaultdict(set)
+        # For each dataset URL, record the indices of the blocks it appears in
+        urls = defaultdict(set)
         for block_index, block in enumerate(cleaned_value):
-            url_path = (
-                block.value.url_path if block.block_type == "dataset_lookup" else urlparse(block.value["url"]).path
-            )
-            url_paths[url_path].add(block_index)
+            if block.block_type == "dataset_lookup":
+                # For dataset lookup blocks, use the website URL
+                url = block.value.website_url
+            else:
+                url = block.value["url"].rstrip("/")  # Treat URLs with and without trailing slashes as equivalent
+            urls[url].add(block_index)
```

Comment:
```suggestion
                url = block.value["url"]  # Treat URLs with and without trailing slashes as equivalent
            urls[.rstrip("/")].add(block_index)
```

so the check is covers manually entered and chosen dataset URLs